crashes while generating secret manifest

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->

**What happened**:

kubectl crashed when trying to generate secret manifest:
```shell
% kubectl create secret generic bastion-k8-secret --from-file=foo=bar -o local/k8-secret.yaml --show-managed-fields=true --dry-run=client
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x536543b]

goroutine 1 [running]:
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/create.(*CreateSecretOptions).Run(0xc00092a800)
        /private/tmp/kubernetes-cli-20211216-32334-1o8gs1i/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/create/create_secret.go:258 +0x2db
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/create.NewCmdCreateSecretGeneric.func1(0xc000930780, {0xc0005289c0, 0x1, 0x6})
        /private/tmp/kubernetes-cli-20211216-32334-1o8gs1i/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/create/create_secret.go:143 +0x74
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc000930780, {0xc000528960, 0x6, 0x6})
        /private/tmp/kubernetes-cli-20211216-32334-1o8gs1i/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:860 +0x5f8
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0003e2280)
        /private/tmp/kubernetes-cli-20211216-32334-1o8gs1i/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /private/tmp/kubernetes-cli-20211216-32334-1o8gs1i/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:902
k8s.io/kubernetes/vendor/k8s.io/component-base/cli.Run(0xc0003e2280)
        /private/tmp/kubernetes-cli-20211216-32334-1o8gs1i/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/component-base/cli/run.go:105 +0x389
main.main()
        _output/local/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:31 +0x1e
```

**What you expected to happen**:

kubectl not to crash, print secret manifest

**How to reproduce it (as minimally and precisely as possible)**:
<!-- Please make sure you are able to reproduce the bug using a supported version and version skew of Kubernetes. See https://kubernetes.io/docs/setup/release/version-skew-policy for which versions and are currently supported.
-->

**Anything else we need to know?**:

**Environment**:
- Kubernetes client and server versions (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:33:37Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.2-eks-06eac09", GitCommit:"5f6d83fe4cb7febb5f4f4e39b3b2b64ebbbe3e97", GitTreeState:"clean", BuildDate:"2021-09-13T14:20:15Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1
```
NB first tried using kubectl 1.22 before installing 1.23
- Cloud provider or hardware configuration:
EKS 1.21
- OS (e.g: `cat /etc/os-release`):
```
% sw_vers   
ProductName:    macOS
ProductVersion: 11.6.2
BuildVersion:   20G314
```

--- a/staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go
@@ -199,13 +199,13 @@ func (o *CreateSecretOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, ar
 
 	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
 	if err != nil {
-		return nil
+		return err
 	}
 
 	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
 	printer, err := o.PrintFlags.ToPrinter()
 	if err != nil {
-		return nil
+		return err
 	}
 
 	o.PrintObj = func(obj runtime.Object) error {

--- a/staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_tls.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_tls.go
@@ -155,13 +155,13 @@ func (o *CreateSecretTLSOptions) Complete(f cmdutil.Factory, cmd *cobra.Command,
 
 	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
 	if err != nil {
-		return nil
+		return err
 	}
 
 	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
 	printer, err := o.PrintFlags.ToPrinter()
 	if err != nil {
-		return nil
+		return err
 	}
 
 	o.PrintObj = func(obj runtime.Object) error {

